src/core/device.c

```diff
@@ -766,7 +773,29 @@ static int device_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 return 0;
         }
 
-        if (streq(action, "remove"))  {
+        if (streq(action, "change"))  {
+                _cleanup_free_ char *e = NULL;
+                Unit *u;
+
+
+                r = unit_name_from_path(sysfs, ".device", &e);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate unit name from device path: %m");
+
+                u = manager_get_unit(m, e);
+                if (!u) {
+                        log_error("Failed to get unit from sysfs name.");
+                        return 0;
+                }
+
+                r = manager_add_job(m, JOB_RELOAD, u, JOB_REPLACE, NULL, NULL);
+                if (r < 0) {
+                        log_error("Failed to add job to manager.");
```

Comment:
always log the error at hand, i.e.:

```c
log_error_errno(r, "Failed to add job to manager: %m");
```

Reply:
no, but it returns negative errno values in its return value, and log_error_errno() accepts "r" as first argument, and %m then maps to that.